ProjFS.Mac/PrjFSKextTests/KauthHandlerTests.mm

```diff
@@ -1,12 +1,50 @@
 #include "../PrjFSKext/kernel-header-wrappers/vnode.h"
 #include "../PrjFSKext/KauthHandlerTestable.hpp"
+#include "../PrjFSKext/VirtualizationRoots.hpp"
+#include "../PrjFSKext/PrjFSProviderUserClient.hpp"
+#include "../PrjFSKext/VirtualizationRootsTestable.hpp"
 #include "../PrjFSKext/PerformanceTracing.hpp"
+#include "../PrjFSKext/public/Message.h"
+#include "../PrjFSKext/ProviderMessaging.hpp"
+#include "../PrjFSKext/public/PrjFSXattrs.h"
 #import <XCTest/XCTest.h>
+#include "KextMockUtilities.hpp"
 #include "MockVnodeAndMount.hpp"
 #include "MockProc.hpp"
 
 using std::shared_ptr;
 
+class PrjFSProviderUserClient
+{
+};
+
+bool ProviderMessaging_TrySendRequestAndWaitForResponse(
+    VirtualizationRootHandle root,
+    MessageType messageType,
+    const vnode_t vnode,
+    const FsidInode& vnodeFsidInode,
+    const char* vnodePath,
+    int pid,
+    const char* procname,
+    int* kauthResult,
+    int* kauthError)
+{
+/*
+   MockCalls::RecordFunctionCall(
```

Comment:
@pmj can you tell what's wrong with the RecordFunctionCall here?  I must be missing something subtle cause it says it doesn't match what's expected.

Once I have this I can validate what we called in the test.

Reply:
Ouch, this is C++ at its worst. I've put a workaround up here: https://github.com/pmj/VFSForGit/tree/more_testing_handlevnodeop - I'll need to go in and finish off the argument checking implementation to know whether this'll work long term. For now, please use the workaround, I'll worry about the fallout. :-)